src/rust/receiver-mock/src/router/otlp.rs

```diff
@@ -130,6 +136,10 @@ pub async fn handler_receiver_otlp_metrics(
             // TODO: Consider giving it some basic capacity to avoid too many allocations.
             let mut samples = vec![];
             for resource_metrics in metrics_data.resource_metrics {
+                if resource_metrics.resource.is_none() {
+                    return HttpResponse::BadRequest().body("No resource info");
+                }
```

Comment:
I'm honestly not sure if BadRequest is the right response. If we get a request without any data, we can just ignore it. In that case, it'd be useful to output a warning, because this is probably not intended. It is valid OTLP though.